serial/serial_test.go

```diff
@@ -56,3 +61,35 @@ func TestOut(t *testing.T) {
 		}
 	}
 }
+
+func TestStartSerial(t *testing.T) {
+	t.Parallel()
+
+	s, err := serial.New(&mockInjector{})
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	restoreMode, err := term.SetRawMode()
+	if err != nil && !errors.Is(err, unix.ENOTTY) {
+		t.Fatal(err)
+	}
```

Comment:
Thanks for this greate test.
SetMode is the terminal's configuration and can be omitted in this test.